src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -1249,17 +1249,87 @@ int InputProcessor::getObjectItemNum(EnergyPlusData &state,
     return getIDFObjNum(state, ObjType, object_item_num); // if incoming input is idf, then return idf object order
 }
 
+void InputProcessor::lowerRangeCheck(EnergyPlusData &state,
```

Comment:
General comment about functions like these. You are passing 9 parameters to this function, 8 of which are only needed in the case of an error, which is presumably rare. This is MCCS (Make the Common Case Slow) 101. If you want to make a function out of the error message code itself, okay. But do the range check at the call site and only call that function if there is an actual error.

Reply:
I went ahead and refactored the function that got split into this one and the other one below, but this only gets used in WeatherManager regardless of whether it was intended to be a general use function. That's another argument for it to go.

Reply:
Yeah. More important than this specific function, however, is the general lesson and pattern. One of the tenets of high-performance (or at least not terrible-performance) programming is "Make the Common Case Fast". Setting up parameters takes time. Don't take that time if those parameters will only be used infrequently, e.g., in the case of an error. Another example I see quite a bit of is functions that have many options that tweak their behavior in different ways. Yes, you are "reusing code" and that's good, but even good things can be taken too far. If those options are rarely used, then you're slowing down the more frequent instances of the function.

These "rules of thumb" are obvious once someone points them out to you. And they are invaluable. I learned this one in my second year graduate computer architecture class. MCCF is *the* rule of thumb for computer hardware design, and it applies to software just as well.